datafusion/sql/src/unparser/plan.rs

```diff
@@ -359,18 +359,14 @@ impl Unparser<'_> {
                             .iter()
                             .map(|e| self.select_item_to_sql(e))
                             .collect::<Result<Vec<_>>>()?;
-                        match &on.sort_expr {
-                            Some(sort_expr) => {
-                                if let Some(query_ref) = query {
-                                    query_ref
-                                        .order_by(self.sort_to_sql(sort_expr.clone())?);
-                                } else {
-                                    return internal_err!(
-                                "Sort operator only valid in a statement context."
-                            );
-                                }
+                        if let Some(sort_expr) = &on.sort_expr {
```

Comment:
Another way I have seen this written is

```suggestion
                        if let Some(sort_expr) = on.sort_expr.as_ref() {
```

Reply:
Ah, yeah - this change was made by `cargo clippy --fix` so I just went with it. I don't really have any preference for which it should be.

Reply:
this is fine too.